logstash-core/lib/logstash/webserver.rb

```diff
@@ -0,0 +1,88 @@
+# encoding: utf-8
+require "puma"
+require 'puma/single'
+require 'puma/binder'
+require 'puma/configuration'
+require 'puma/commonlogger'
+
+module LogStash 
+  class WebServer
+
+    extend Forwardable
+
+    attr_reader :logger, :status, :config, :options, :cli_options, :runner, :binder, :events
+
+    def_delegator :@runner, :stats
+
+    def initialize(logger, options={})
+      @logger      = logger
+      http_port    = options[:http_port] || 9600
+      @options     = {}
+      @cli_options = options.merge({ :rackup => ::File.join(::File.dirname(__FILE__), "api", "init.ru"), 
+                                     :binds => ["tcp://0.0.0.0:#{http_port}"] })
+      @status      = nil
+
+      parse_options
+
+      @runner  = nil
+      @events  = ::Puma::Events.strings
+      @binder  = ::Puma::Binder.new(@events)
+      @binder.import_from_env
+
+      set_environment
+    end
+
+    def run
+      log "=== puma start: #{Time.now} ==="
+
+      @runner = Puma::Single.new(self)
+      @status = :run
+      @runner.run
+      stop(:graceful => true)
+    end
+
+    def log(str)
+      logger.debug(str) if logger.debug?
```

Comment:
I don't see a big point in checking `if logger.debug?` at this point. the `str` argument has already been evaluated and the cost has been paid for it